lib/Models/BaseMaps/BaseMapsModel.ts

```diff
@@ -52,20 +52,17 @@ export class BaseMapsModel extends CreateModel(BaseMapsTraits) {
   get baseMapItems(): BaseMapItem[] {
     const enabledBaseMaps: BaseMapItem[] = [];
 
-    this.items.forEach((baseMapItem) => {
-      if (
-        baseMapItem.item &&
-        !ModelReference.isRemoved(baseMapItem.item) &&
-        (!this.enabledBaseMaps ||
-          this.enabledBaseMaps.includes(baseMapItem.item))
-      ) {
-        const itemModel = this.terria.getModelById(BaseModel, baseMapItem.item);
-        if (MappableMixin.isMixedInto(itemModel))
+    this.enabledBaseMaps.forEach((baseMapItem) => {
```

Comment:
`this.enabledBaseMaps` might be undefined, and when undefined it should show all base maps available. 
i.e. simple sorting before return in the old code would do the trick 
```ts
// Sort based on enabledBaseMaps order
if (this.enabledBaseMaps) {
  enabledBaseMaps.sort((a, b) => {
    const aIndex = this.enabledBaseMaps!.indexOf(a.item.uniqueId!);
    const bIndex = this.enabledBaseMaps!.indexOf(b.item.uniqueId!);
    return aIndex - bIndex;
  });
}
```

or something little bit cleaner with utility function
```ts
const createBaseMapItem = (baseMapItem: BaseMapModel): BaseMapItem | undefined => {
  if (baseMapItem.item && !ModelReference.isRemoved(baseMapItem.item)) {
    const itemModel = this.terria.getModelById(BaseModel, baseMapItem.item);
    if (MappableMixin.isMixedInto(itemModel)) {
      return {
        image: baseMapItem.image,
        contrastColor: baseMapItem.contrastColor,
        item: itemModel
      };
    }
  }
};

if (!this.enabledBaseMaps) {
  return this.items
    .map(createBaseMapItem)
    .filter((item): item is BaseMapItem => item !== undefined);
}

return this.enabledBaseMaps
  .map(enabledId => {
    const baseMapItem = this.items.find(item => item.item === enabledId);
    return baseMapItem && createBaseMapItem(baseMapItem);
  })
  .filter((item): item is BaseMapItem => item !== undefined);
```

Reply:
on first look, it should work correctly and the rest is a personal preference on the code structure. It would be great to cover sorting with the unit test to avoid regression bugs in future, there are tests for BaseMapsModel 

https://github.com/TerriaJS/terriajs/blob/2490260c78e4b7666496bfc6d0ec677eac63a98c/test/Models/BaseMaps/BaseMapsModelSpec.ts